### PR TITLE
Delete Topics

### PR DESCRIPTION
**Pull request by noahtallen**
One can now delete topics in the admin portal. I had to make a few changes to topic code:
1. Get the ID of the topic for their image - so create the doc before uploading
2. Save the imageURL back to the database after uploading
3. Utilize the ID for deletion and for the react `key` for the topic cards.
4. I noticed some unnecessary code in `index.ts` - the `onSnapshot` gets the data when it's first called so we don't need to get the data first. I also changed how the handler is called very slightly, and I think it's fine but maybe @JakeThurman had a specific reason for it (i'll make a comment in the code)

Also made some QOL improvements such as:
1. Readme with instructions for deployment and development
2. Package commands for running locally and deploying
3. Removed some Firestore files that could cause issues if not used properly. These are configured online, so it isn't an issue to remove them here.

### CODE REVIEW

File: web-dev/index.ts [hunk 1]
```diff
@@ -8,22 +8,20 @@ window["firebase"] = firebase;
...
+            data.id = d.id
+            return data
+        }));
 
-    // Call now and listen for changes
-    ref.get().then(useCallback).catch(
-        console.log
-    );
-    return ref.onSnapshot({ next: useCallback });
+    return ref.onSnapshot(useCallback);
```
[the remark this answers is not in the document]
Comment (JakeThurman) [reply web-dev/index.ts]: my typings said to do that, so I did haha

[the remark this answers is not in the document]
Comment (noahtallen) [reply web-dev/index.ts]: 😆That's funny lol. For some reason it worked - you normally call it like:
```js
ref.onSnapshot(snap => {
  // etc
})
```
so you can pass the callback right in

**Comment by JakeThurman** [comment]
LGTM

**Comment by JakeThurman** [comment]
Thanks for doing this noah

**Pull request merged**